test/ace/http1/server_test.exs

```diff
@@ -608,6 +608,89 @@ defmodule Ace.HTTP1.ServerTest do
     assert 3 == state.count
   end
 
+  test "Request backpressure kicks in if the worker doesn't handle request parts quickly enough" do
+    {:ok, service} =
+      Ace.HTTP.Service.start_link(
+        {Raxx.Forwarder, %{target: self()}},
+        port: 0,
+        cleartext: true
+      )
+
+    {:ok, port} = Ace.HTTP.Service.port(service)
+
+    # oversize content-length so that we don't make assumptions about the tcp buffer size
+    request_head = """
+    POST / HTTP/1.1
+    host: example.com
+    content-length: 20000000
+
+    """
+
+    socket_opts = [:binary, {:send_timeout, 100}, {:send_timeout_close, false}]
+    {:ok, socket} = :gen_tcp.connect({127, 0, 0, 1}, port, socket_opts)
+
+    :ok = :gen_tcp.send(socket, request_head)
+
+    assert_receive {:"$gen_call", from, {:headers, _request, state}}, 1000
+    GenServer.reply(from, {[], state})
+    one_kb_message = String.duplicate("bla ", div(1024, 4))
+
+    sent_message_count = fill_up_buffer(socket, one_kb_message, 10_000)
```

Comment:
Why so many? could this test work if we sent just 11 packets?

Reply:
I don't think so, IIRC on my machine you could push ~450 1kb messages before `:gen_tcp.send` returned a timeout. It seems like a lot to me too.

The way I see it all stems from filling the buffers in lower layers, so the buffer size in `:gen_tcp` (on both the sending and receiving size) and the OS buffer. We could probably push it way down if we tweaked `:inet` and `:gen_tcp` options, but it would make the test more complicated, obscure its purpose a bit and still not probably bring it down to 11 packets.

The Erlang network stack is still a bit of a black box to me, I'd be curious myself where these bytes go to in the end and how you can tweak it.